modules/adbookpspBidAdapter.js

```diff
@@ -0,0 +1,768 @@
+  let videoObj = {
+    w,
+    h,
+  };
+
+  for (const param of VIDEO_PARAMS) {
+    const paramsValue = utils.deepAccess(bidRequest, `params.video.${param}`);
+    const mediaTypeValue = utils.deepAccess(bidRequest, `mediaTypes.video.${param}`);
+
+    if (paramsValue || mediaTypeValue) {
+      videoObj[param] = paramsValue || mediaTypeValue;
+    }
+  }
+
+  return videoObj;
+}
+
+function buildImpExt(validBidRequest) {
+  const defaultOrgId = getBidderConfig('orgId');
+  const { orgId, placementId } = validBidRequest.params || {};
+  const effectiverOrgId = orgId || defaultOrgId;
+  const ext = {};
+
+  if (placementId) {
+    utils.deepSetValue(ext, 'adbook.placementId', placementId);
+  }
+
+  if (effectiverOrgId) {
+    utils.deepSetValue(ext, 'adbook.orgId', effectiverOrgId);
+  }
+
+  return ext;
+}
+
+/**
+ * BID RESPONSE
+ */
+
+function interpretResponse(bidResponse, bidderRequest) {
+  const bidderRequestBody = safeJSONparse(bidderRequest.data);
+
+  if (
+    utils.deepAccess(bidderRequestBody, 'id') !=
+    utils.deepAccess(bidResponse, 'body.id')
+  ) {
+    utils.logError(
+      `${BIDDER_CODE}: Bid response id does not match bidder request id`
+    );
+
+    return [];
+  }
+
+  const referrer = utils.deepAccess(bidderRequestBody, 'site.ref', '');
+  const incomingBids = utils
+    .deepAccess(bidResponse, 'body.seatbid', [])
+    .filter((seat) => utils.isArray(seat.bid))
+    .reduce((bids, seat) => bids.concat(seat.bid), [])
+    .filter(validateBid(bidderRequestBody));
+  const targetingMap = buildTargetingMap(incomingBids);
+
+  return impBidsToPrebidBids(
+    incomingBids,
+    bidderRequestBody,
+    bidResponse.body.cur,
+    referrer,
+    targetingMap
+  );
+}
+
+function impBidsToPrebidBids(
+  incomingBids,
+  bidderRequestBody,
+  bidResponseCurrency,
+  referrer,
+  targetingMap
+) {
+  return incomingBids
+    .map(
+      impToPrebidBid(
+        bidderRequestBody,
+        bidResponseCurrency,
+        referrer,
+        targetingMap
+      )
+    )
+    .filter((i) => i !== null);
+}
+
+const impToPrebidBid =
+  (bidderRequestBody, bidResponseCurrency, referrer, targetingMap) => (bid) => {
+    try {
+      const bidRequest = findBidRequest(bidderRequestBody, bid);
+
+      if (!bidRequest) {
+        utils.logError(`${BIDDER_CODE}: Could not match bid to bid request`);
+
+        return null;
+      }
+      const categories = utils.deepAccess(bid, 'cat', []);
+      const mediaType = getMediaType(bid.adm);
+      let prebidBid = {
+        ad: bid.adm,
+        adId: bid.adid,
+        adserverTargeting: targetingMap[bid.impid],
+        adUnitCode: bidRequest.tagid,
+        bidderRequestId: bidderRequestBody.id,
+        bidId: bid.id,
+        cpm: bid.price,
+        creativeId: bid.crid || bid.id,
+        currency: bidResponseCurrency || getBidderConfig('defaultCurrency'),
+        height: bid.h,
+        lineItemId: utils.deepAccess(bid, 'ext.liid'),
+        mediaType,
```

Comment:
is this supposed to be mediaType: mediaType, ?

Reply:
Actually, on a second thought, since I use this notation in multiple places and it is being transpiled anyway, I'd like to leave it as is if that's ok with you.